app/workers/asset_manager_create_attachment_asset_worker.rb

```diff
@@ -0,0 +1,51 @@
+class AssetManagerCreateAttachmentAssetWorker < WorkerBase
+  include AssetManager::ServiceHelper
+
+  # Carrierwave runs on an after_save hook and the transaction that inserts Assetable into the database
+  # might not be committed yet. This can cause a race condition where the worker runs before the assetable is readable.
+  # Use TransactionAwareClient for this worker to ensure that the commit is finished before the worker is executed.
+  sidekiq_options queue: "asset_manager", client_class: Sidekiq::TransactionAwareClient
+
+  def perform(temporary_location, asset_params, attachable_model_class, attachable_model_id, draft = false, auth_bypass_ids = [])
+    return unless File.exist?(temporary_location)
+
+    attachment_data = AttachmentData.where(id: asset_params["assetable_id"]).first
+
+    return logger.info("Assetable AttachmentData of id #{asset_params['assetable_id']} does not exist") if attachment_data.nil?
+
+    attachable = attachable_model_class.constantize.where(id: attachable_model_id).first
+    if attachable.nil?
+      return logger.info("Attachable #{attachable_model_class} of id #{attachable_model_id} does not exist")
+    end
+
+    file = File.open(temporary_location)
+
+    asset_options = { file:, auth_bypass_ids:, draft: }
+    if attachable.respond_to?(:access_limited?) && attachable.access_limited?
+      asset_options[:access_limited_organisation_ids] = AssetManagerAccessLimitation.for(attachable)
+    end
+
+    response = create_asset(asset_options)
+
+    file.close
+    FileUtils.rm(file)
+    FileUtils.rmdir(File.dirname(file))
+
+    asset = attachment_data.assets.where(variant: asset_params["asset_variant"]).first_or_initialize
+    asset.asset_manager_id = response.asset_manager_id
+    asset.filename = response.filename
+    asset.save!
+
+    if attachable.is_a?(Edition)
+      PublishingApiDraftUpdateWorker.perform_async(attachable.class.to_s, attachable.id)
+    else
+      # At present we can only reach this branch when the attachable is a PolicyGroup. No other non-editionable models have attachments.
```

Comment:
I still believe we should nonetheless update the `publishing-api` here too. That'd be the most consistent thing to do, and addresses the case below: 

The user may save the document with attachments still pending, meaning they would be filtered out at the presenter level (`WorkingGroupPresenter` -> uses default filtering in `govspeak_helper.rb`) so the document would be sent without any attachments. In typical editions we have added validation so that the user would be informed they cannot publish as attachments are still processing. `PolicyGroup` is the outlier since it uses the `PublishesToPublishingApi` concern flow. 

I'd prefer it if we followed up with a publishing-api call to republish the group, making sure the correct urls are sent here if they haven't already.

Reply:
By "attachments still pending" do you mean "not yet uploaded to Asset Manager"?

Reply:
Also I don't think that would work because we'd have to push a draft and call publish in Publishing API to handle that, and generally we shouldn't be calling publish here

Reply:
Trying to  have attachments with drafts and policy groups without is a flawed model, though

Reply:
With the "PROCESSING" tag, yes.

Reply:
> By "attachments still pending" do you mean "not yet uploaded to Asset Manager"?

With the "PROCESSING" tag, yes.

Reply:
> Also I don't think that would work because we'd have to push a draft and call publish in Publishing API to handle that, and generally we shouldn't be calling publish here

I was thinking implementing the `republish_on_assets_ready` equivalent for the `PolicyGroup` too.